app/src/main/java/org/sopt/sample/presentation/signin/view/SignInActivity.kt

```diff
@@ -51,53 +44,73 @@ class SignInActivity : AppCompatActivity() {
             this, EventObserver { isPossible ->
                 if (isPossible) {
                     Toast.makeText(this, R.string.success_sign_in, Toast.LENGTH_SHORT).show()
-                    setSharedPreferenceToUser(viewModel.getUser()!!)
                     startMainActivity()
                 } else {
                     Toast.makeText(this, R.string.failure_sign_in, Toast.LENGTH_SHORT).show()
                 }
             }
         )
+        viewModel.isPromising.observe(
+            this, EventObserver { isSuccess ->
+                if (isSuccess) {
+                    binding.signInSignInBtn.isEnabled = true
+                    binding.signInSignInBtn.backgroundTintList =
+                        ColorStateList.valueOf(
+                            ContextCompat.getColor(
+                                this,
+                                R.color.black
+                            )
+                        )
+                } else {
+                    binding.signInSignInBtn.isEnabled = false
+                    binding.signInSignInBtn.backgroundTintList =
+                        ColorStateList.valueOf(
+                            ContextCompat.getColor(
+                                this,
+                                R.color.gray
+                            )
+                        )
+                }
+            }
+        )
     }
 
-    private fun setActivityResultLauncher() {
-        launcher =
-            registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { result ->
-                if (result.resultCode == RESULT_OK) {
-                    val user: User? = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
-                        result.data?.getSerializableExtra(USER, User::class.java)
-                    } else {
-                        result.data?.getSerializableExtra(USER) as User?
-                    }
-                    Snackbar.make(binding.root, R.string.success_sign_up, Snackbar.LENGTH_SHORT)
-                        .show()
-                    viewModel.setUser(user!!)
+    private fun setOnClickListener() {
+        with(binding) {
+            signInSignInBtn.setOnClickListener {
+                viewModel.signIn(getUser())
+            }
+            signInSignUpBtn.setOnClickListener {
+                startSignUpActivity()
+            }
+        }
+    }
+
+    private fun setAddTextChangedListener() {
+        with(binding) {
+            signInIdEt.addTextChangedListener {
+                if (signInIdEt.text.isNullOrEmpty()) {
+                    viewModel.setUserStatus(EMAIL, false)
+                } else {
+                    viewModel.setUserStatus(EMAIL, true)
+                }
```

Comment:
불필요한 조건식은 없애주셔도 괜찮을 것 같아요!

```suggestion
                viewModel.setUserStatus(EMAIL, !signInIdEt.text.isNullOrEmpty())
```